src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp

```diff
@@ -2357,7 +2358,7 @@ void C2_MacroAssembler::string_equals_v(Register a1, Register a2, Register resul
 
   mv(result, false);
 
-  element_compare(a1, a2, result, cnt, tmp1, tmp2, v2, v4, v2, true, DONE);
+  element_compare(a1, a2, result, cnt, tmp1, tmp2, v2, v4, v2, true, DONE, Assembler::m2);
```

Comment:
It's not apperent to me why you use m2 here. Can you add a comment?

Reply:
Agree. I will verify the string equals later soon, in that patch, I will add the comment accordingly based on test result. Here, it's the previous value which I just passed in as explicit value.

Reply:
I'd like to discuss another thing related to this. Currently, in riscv_v.ad file, we explicitly ask for the temp v registers we need, but in corresponding macro assembler code, there is no such explicit statement about it. So there is kind of a risk here, i.e. a future patch could accidentally change the code from `element_compare(a1, a2, result, cnt, tmp1, tmp2, v2, v4, v2, true, DONE, Assembler::m2);` to `element_compare(a1, a2, result, cnt, tmp1, tmp2, v4, v8, v4, true, DONE, Assembler::m4);`, on the other hand it could fail to change corresponding code in riscv_v.ad file, i.e forget to reserve the v6-v11.
I think it helps to make all such calls (i.e. vector code that use more than one v register in one register group) explicit, i.e. in riscv_v.ad file, we pass all the temporary vector registers explicitly when calling into functions in macroAssemble functions. It will make the code more tedious, but it’s safer than before, else it’s error-prone, and the potential bugs will be random. Of course, I will put the patch in a new pr to address all existing such code.
How do you think about it? Maybe there is other better way to address such issue? Please kindly share your suggestion.

Reply:
Maybe we can create a class for vector register groups and pass references instead? That would help eliminate the problem of passing too many parameters for vector registers.

Reply:
Generally speaking, it would be great for C2 to be more register-group-aware. So if we specify `vReg_V2m2` it includes both `v2` and `v3` for example, and marking a `vReg_V2m2` as TEMP marks both `v2` and `v3` as TEMP implicitly. 

We already have the concept of `operand` ([`vRef_V2`](https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/riscv/riscv.ad#L3439-L3447) for example), but that expresses a `vRef_V2` is any of the register in the list `[v2_reg]` (aka `v2_reg` only). I don't know that there is a concept for a `vRef_V2m2` being a group of register (aka all register in `[v2_reg, v3_reg]`).

Reply:
I can do further investigation on it, I kindly like what @luhenry suggested above, if we can implement it. 
But anyway I think it will be in another pr.

Reply:
Here https://github.com/openjdk/jdk/pull/20040 is the prototype version, just want to get some feedback about the interface & design, i.e. what it looks like, rather than a normal review (Of course, if you have any concerns about its implementation, also welcome to point it out in advance.).
If we are happy with this interface, I will investigate further and implement it.
BTW, currently this prototype only compiles successfully, can not run yet.